### DISCUSSION TITLE
Pass different env variables to different entries

### DISCUSSION THREAD

**Discussion by xblurx** [webpack/webpack#15345] [Show and tell]
Hi there, does anyone know is there a way to pass different instances of a same environment variable to different entries when bundling? The problem is, say, I have this structure
```
const entry = {
        'entry1': './src/entry1',
        'entry2': './src/entry2',
    };
```
and each of these modules references an `appVersion` variable, that needs to resolve differently in case of each entry.
I suppose, something like this would have worked
```
const entry = {
        'entry1': {
            import: './src/entry1',
            env: {appVersion: '01.000.00'}
          }
        'entry2': {
            import: './src/entry2',
            env: {appVersion: '01.001.00'}
          }
    };
```
but seems like Entry API is no help here

**Comment by alexander-akait** [comment]
Sorry, it was not design for this, you can use different ENV variables, i.e. `ENTRY_1_APP_VERSION` and `ENTRY_2_APP_VERSION`

**Reply by xblurx** [reply]
the thing is, creating 2 diff env variables is useless in this case, cause in the end portion of code that is shared between entries only references 1 same variable

**Reply by alexander-akait** [reply]
Sorry, it is impossible
